packages/core/saved-objects/core-saved-objects-migration-server-internal/src/model/helpers.ts

```diff
@@ -285,3 +286,13 @@ export function getMigrationType({
  */
 export const getTempIndexName = (indexPrefix: string, kibanaVersion: string): string =>
   `${indexPrefix}_${kibanaVersion}_reindex_temp`;
+
+/** Increase batchSize by 20% until a maximum of defaultBatchSize */
+export const increaseBatchSize = (
+  stateP: OutdatedDocumentsSearchRead | ReindexSourceToTempRead
+) => {
+  const increasedBatchSize = Math.floor(stateP.batchSize * 1.2);
+  return increasedBatchSize > stateP.defaultBatchSize
+    ? stateP.defaultBatchSize
+    : increasedBatchSize;
```

Comment:
NIT: Same remark as @gsoldevila , `defaultBatchSize` seems a bit misleading given it's both the *default* and the *maximum*.

Yeah, naming is hard.